extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/StandardSecurityCheckInterceptor.java

```diff
@@ -36,9 +38,15 @@ public abstract class StandardSecurityCheckInterceptor {
     public Object intercept(InvocationContext ic) throws Exception {
         // RoutingContext can be null if RESTEasy is used together with other stacks that do not rely on it (e.g. gRPC)
         // and this is not invoked from RESTEasy route handler
-        if (controller.isAuthorizationEnabled() && currentVertxRequest.getCurrent() != null) {
-            Method method = currentVertxRequest.getCurrent().get(EagerSecurityFilter.class.getName());
-            if (method != null && method.equals(ic.getMethod())) {
+        RoutingContext routingContext = currentVertxRequest.getCurrent();
+        if (controller.isAuthorizationEnabled() && routingContext != null) {
+            Method method = routingContext.get(EagerSecurityFilter.class.getName());
+            if (method == null) {
+                // if this interceptor is run on resource method it means this is parent method for subresource
```

Comment:
I don't quite understand this comment, please clarify

Reply:
Sadly in RESTEasy Classic the `jakarta.ws.rs.container.ContainerRequestFilter` is only applied once. When there is parent resource and sub-resource, the filter is only applied on sub-resource.

In RESTEasy Reactive the `ServerRestHandler` is applied on every endpoint (even the parent resource that provides sub-resource), which allows eager security be performed in the filter on parent resource (depending on where RBAC is placed).

This PR doesn't change this state - security check will be applied in the same moment it is till now, I just needed to find a way to say that check has happen. This is not right place to discuss this, please use IM.

Reply:
hmm, maybe I can be even clearer, CDI interceptors for parent endpoint providing Jakarta REST sub-resource, therefore in that case,  CDI interceptor is performing eager security. I was probably too caution in describing this because I didn't like it. But there is nothing fishy, it's safe as it runs before serialization.

Reply:
@michalvavrik Can you give me a favor and type the endpoint code that this check is dealing with and clarify how where it was dealt with in the old code, just so that I fully follow, thanks

Reply:
The point of this exact comment you are pointing at is that nothing changed. CDI interceptor of parent resource annotated with RBAC performed it in past and still performs it. What I did here is to propagate information that check has been performed to Jakarta REST filter so that we know that default JAX-RS security should be applied. The default JAX-RS security is only applied when parent resource is not secured. As you can see I didn't change any test, therefore contract sticks.

The comment is related to 2 tests: 

- `io.quarkus.resteasy.test.security.DefaultRolesAllowedJaxRsTest#shouldAllowPermitAllSubResource`
- `io.quarkus.resteasy.test.security.DefaultRolesAllowedJaxRsTest#shouldAllowPermitAllClass`

In short: you have parent resource with RBAC

```
@Path("/permitAll")
@PermitAll
public class PermitAllResource {

    @Path("/sub")
    public UnsecuredSubResource sub() {
        return new UnsecuredSubResource();
    }

}
```
and you have sub-resource

```
public class UnsecuredSubResource {
    @GET
    @Path("/subMethod")
    public String subMethod() {
        return "subMethod";
    }
}
```

here, default JAX-RS security must not be applied as parent resource is already _secured_ (well, you know...). I didn't write these contracts and FYI they are same in RESTEasy Reactive. `jakarta.ws.rs.container.ContainerRequestFilter` is only applied after `sub` has been invoked and secured with CDI, however no serialization is in place, hence safe. Just try `mvn clean verify -f extensions/resteasy-classic/resteasy/deployment/ -Dtest=DefaultRolesAllowedJaxRsTest#shouldAllowPermitAllClass -Dmaven.surefire.debug`

Reply:
@michalvavrik Thanks for the explanation.